modules/overview-of-data-science-pipelines-caching.adoc

```diff
@@ -27,35 +27,110 @@ Cached steps in pipelines are visually indicated in the user interface (UI):
 
-== Disabling caching for specific tasks or pipelines
+== Disabling caching
 
-In {productname-short}, caching is enabled by default, but there are situations where disabling caching for specific tasks or the entire pipeline is necessary. For example, tasks that rely on frequently updated data or unique computational needs might not benefit from caching.
+ifdef::upstream,self-managed[]
+In {productname-short}, caching is enabled by default to improve performance, but there are situations where you might want to disable caching for specific tasks, an entire pipeline, or all pipelines. For example, caching might not be beneficial for tasks that rely on frequently updated data or unique computational needs. In other cases, such as debugging, development, or when deterministic re-execution is required, you might want to disable caching for all pipelines.
+
+[CAUTION]
+====
+Disabling caching at the pipeline or pipeline server level causes all tasks to run again, potentially increasing compute time and resource usage.
+====
+endif::[]
+
+ifdef::cloud-service[]
+In {productname-short}, caching is enabled by default to improve performance, but there are situations where you might want to disable caching for specific tasks or the entire pipeline. For example, caching might not be beneficial for tasks that rely on frequently updated data or unique computational needs. 
+
+[CAUTION]
+====
+Disabling caching at the pipeline server level causes all tasks to run again, potentially increasing compute time and resource usage.
+====
+endif::[]
+
+You can control caching for data science pipelines in the following ways:
+
+* *Individual task*: Data scientists can disable caching for specific steps in a pipeline.
+* *Pipeline (submit time)*: Data scientists can disable caching when submitting a pipeline run.
+* *Pipeline (compile time)*: Data scientists can disable caching when compiling a pipeline.
+ifdef::upstream,self-managed[]
+* *All pipelines (pipeline server)*: Cluster administrators can disable caching for all pipelines in the pipeline server.
+endif::[]
 
 === Disabling caching for individual tasks
 
 To disable caching for a particular task, apply the `set_caching_options` method directly to the task in your pipeline code:
 
 `task_name.set_caching_options(False)`
 
-After applying this setting, {productname-short} executes the task in all future pipeline runs, ignoring any cached results.
+After applying this setting, {productname-short} runs the task in future pipeline runs, ignoring any cached results.
 
 [NOTE]
 ====
-You can re-enable caching for individual tasks by setting `set_caching_options(True)`.
+You can re-enable caching for individual tasks by setting `set_caching_options(True)`. This setting is ignored if caching is disabled in the pipeline server.
```

Comment:
Shouldn't `This setting is ignored if caching is disabled in the pipeline server.` be present only when `ifdef::upstream,self-managed[]`?